### PR TITLE
hotspot add mod:CLOSE;MONITOR;CONTROL

### PR DESCRIPTION
**Pull request by cafra**
### Describe what this PR does / why we need it

Add hotspot rule mod field for meeting the needs of monitoring mode

### Does this pull request fix one issue?

NO

### Special notes for reviews

### CODE REVIEW

File: core/hotspot/rule.go [hunk 1]
```diff
@@ -97,8 +97,20 @@ type Rule struct {
 	ParamsMaxCapacity int64 `json:"paramsMaxCapacity"`
 	// SpecificItems indicates the special threshold for specific value
 	SpecificItems map[interface{}]int64 `json:"specificItems"`
+	// Mod Switch
```
Comment (ansiz) [inline core/hotspot/rule.go:100]: Mod -> Mode

File: core/hotspot/rule_manager.go [hunk 2]
```diff
@@ -154,6 +154,9 @@ func onRuleUpdate(rules []*Rule) (err error) {
 			logging.Warn("[HotSpot onRuleUpdate] Ignoring invalid hotspot rule when loading new rules", "rule", r, "err", err.Error())
 			continue
 		}
+		if r.Mode == "" {
+			r.Mode = CLOSE
```
Comment (ansiz) [inline core/hotspot/rule_manager.go:158]: Mode == ""或者CLOSE时，是不是应该直接不放这条规则到RuleMap里

File: core/hotspot/slot.go [hunk 3]
```diff
@@ -52,18 +52,27 @@ func (s *Slot) Check(ctx *base.EntryContext) *base.TokenResult {
 	result := ctx.RuleCheckResult
 	tcs := getTrafficControllersFor(res)
 	for _, tc := range tcs {
+		if tc.BoundRule().Mode == CLOSE {
```
Comment (ansiz) [inline core/hotspot/slot.go:55]: 如果前面为CLOSE就没放到Map里，这里就不需要这个判断了吧

File: core/hotspot/rule.go (hunk 1 above)
Comment (cafra) [reply core/hotspot/rule.go:100]: ok

File: core/hotspot/slot.go (hunk 3 above)
Comment (cafra) [reply core/hotspot/slot.go:55]: 加载时忽略close rule

File: core/hotspot/rule_manager.go (hunk 2 above)
Comment (cafra) [reply core/hotspot/rule_manager.go:158]: 去掉了

**Review by sczyh30 — APPROVED** [review]

**Pull request merged**